Sources/Shared/Requests/Request+ConnectedApp.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  Request+ConnectedApp.swift
+//  VimeoNetworking
+//
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+public extension Request {
+
+    /// Returns a fetch request of all connected apps for the authenticated user.
+    static func connectedApps() -> Request {
+        return Request(path: .connectedAppsURI)
+    }
+
+    /// Returns a fetch request for a single connected app.
+    /// - Parameter appType: The `ConnectedAppType` to fetch data for.
+    static func connectedApp(_ appType: ConnectedAppType) -> Request {
+        return Request(path: .connectedAppsURI + appType.stringValue)
+    }
+
+    /// Returns a `put` request for connecting the provided app type to the current authenticated user's account.
+    /// - Parameters:
+    ///   - appType: The app platform for which the connection will be established.
+    ///   - token: An authentication token from the provided platfrom, used to establish the connection.
+    static func connect(to appType: ConnectedAppType, with token: String) -> Request {
```

Comment:
Is it worth making this request more generic? As in do we always expect a token to be the method of authenticating with a third party? Could there be cases where we need to pass more than 1 parameter?

Reply:
For the platforms we support we only require the token for authentication. I guess it's possible that some future platform might have different requirements? But I think for now this should be fine.